File: algorithms/1995.count-special-quadruplets/solution.go
func countQuadruplets(nums []int) int {
	l := len(nums)
	count := 0

	for a := 0; a < l-3; a++ {
		for b := a + 1; b < l-2; b++ {
			for c := b + 1; c < l-1; c++ {
				for d := c + 1; d < l; d++ {
					if nums[a]+nums[b]+nums[c] == nums[d] {
						count++
					}
				}
			}
		}
	}

	return count
}

File: algorithms/1995.count-special-quadruplets/solution.rs
impl Solution {
    pub fn count_quadruplets(nums: Vec<i32>) -> i32 {
        let len = nums.len();
        let mut count = 0;

        for a in 0..(len - 3) {
            for b in (a + 1)..(len - 2) {
                for c in (b + 1)..(len - 1) {
                    for d in (c + 1)..len {
                        if nums[a] + nums[b] + nums[c] == nums[d] {
                            count += 1;
                        }
                    }
                }
            }
        }

        count
    }
}
